cas-server-integration-hazelcast/src/main/java/org/jasig/cas/ticket/registry/HazelcastTicketRegistry.java

```diff
@@ -130,8 +132,40 @@ public Ticket getTicket(final String ticketId) {
      */
     @Override
     public boolean deleteTicket(final String ticketId) {
-        logger.debug("Removing ticket [{}]", ticketId);
-        return this.registry.remove(ticketId) != null;
+        if (ticketId == null) {
+            return false;
+        }
+
+        final Ticket ticket = getTicket(ticketId);
+        if (ticket == null) {
+            return false;
+        }
+
+        if (ticket instanceof TicketGrantingTicket) {
```

Comment:
There is a fair amount of duplicate code here, I suppose, across all registries. I think we need to clean this all up later, but certainly not in this PR.